src/functional.jl

```diff
@@ -0,0 +1,314 @@
+# Generate a functional interface from the Assertion and Imputor types.
+"""
+    splitkwargs(::Type{T}, kwargs...) where T -> (imp, rem)
+
+Takes a type with kwargs and returns the constructed type and the
+unused kwargs which should be passed to the `impute!` call.
+
+NOTE: This is only intended to be used internally
+"""
+function splitkwargs(::Type{T}, kwargs...) where T
+    rem = Dict(kwargs...)
+    kwdef = empty(rem)
+
+    for f in fieldnames(T)
+        if haskey(rem, f)
+            kwdef[f] = rem[f]
+            delete!(rem, f)
+        end
+    end
+
+    return (T(; kwdef...), rem)
+end
+
+const global assertion_methods = (
+    threshold = Threshold,
+)
+
+const global imputation_methods = (
+    drop = DropObs,
+    dropobs = DropObs,
+    dropvars = DropVars,
+    interp = Interpolate,
+    interpolate = Interpolate,
+    fill = Fill,
+    locf = LOCF,
+    nocb = NOCB,
+    srs = SRS,
+    svd = SVD,
+    knn = KNN,
+)
+
+for (f, v) in pairs(assertion_methods)
+    typename = nameof(v)
+    @eval begin
+        function $f(data; kwargs...)
+            a, rem = splitkwargs($typename, kwargs...)
+            return assert(data, a; rem...)
+        end
+        function $f(; kwargs...)
+            a, rem = splitkwargs($typename, kwargs...)
+            return data -> assert(data, a; rem...)
+        end
+    end
+end
+
+for (f, v) in pairs(imputation_methods)
+    typename = nameof(v)
+    f! = Symbol(f, :!)
+
+    @eval begin
+        function $f(data; kwargs...)
+            imp, rem = splitkwargs($typename, kwargs...)
+            return impute(data, imp; rem...)
+        end
+        function $f!(data; kwargs...)
+            imp, rem = splitkwargs($typename, kwargs...)
+            return impute!(data, imp; rem...)
+        end
+        function $f(; kwargs...)
+            imp, rem = splitkwargs($typename, kwargs...)
+            return data -> impute(data, imp; rem...)
+        end
+        function $f!(; kwargs...)
+            imp, rem = splitkwargs($typename, kwargs...)
+            return data -> impute!(data, imp; rem...)
+        end
```

Comment:
Do we have to have these curried methods?

Reply:
I suppose not, I just like them over explicitly constructing the types. You're probably right though, keeping the options down to either `Impute.<method>(data; kwargs...)` or `imp = Imputor(...); ... impute(data, imp; kwargs...)` is probably prudent.

Reply:
Deprecated